index.js

```diff
@@ -35,44 +52,43 @@ app.get('/api/phonenumbers/parse/text/:pString', function (req, res) {
   phone = phoneUtil.parse(inString, 'CA');
   phone = phoneUtil.format(phone, PNF.INTERNATIONAL);
   res.send(phone);
-  
+
 
 });
 
-app.post('/api/phonenumbers/parse/file', upload.single('file'), function (req, res) {
-  
-   if (!req.file) {
+app.post('/api/phonenumbers/parse/file', upload.single('file'), async function (req, res) {
+
+  if (!req.file) {
     res.status(400).send('File not found');
-  } 
-   else {
+  }
+
+  else {
     var inFile = req.file.path;
-    
-
-    fs.readFile(req.file.path, function (err, contents) {
-      if (err) {
-        res.status(500).send(err);
-        return;
-      }
-      var fileText = contents.toString('ascii');
-      var buf = Buffer.from(fileText, 'base64');
-      var numbers = buf.toString('ascii');
-      var numArr = numbers.split('\n');
-      var phone = [];
-      for(var i =0; i<numArr.length; i++){
-        phone[i] = numArr[i];
-        phone[i] = phoneUtil.parse(numArr[i], 'CA');
-        phone[i]= phoneUtil.format(phone[i], PNF.INTERNATIONAL);
-      }
-
-      res.status(200).send(phone);
-    });
+
+    // Check if file is PDF
+    if (req.file.originalname.match(/.*.pdf/)) {
+
+      // Use pdfText to read the PDF file appropriately
+      pdfText(inFile, function (err, chunks) {
+        res.status(200).send(parseOutput(chunks));
+      });
+
+    }
+    else {
```

Comment:
This part of the code does not work to the specifications of the original assignment where it converts a base64 converted text. Please use my original code with your PDF reader function.